src/gentropy/l2g.py

```diff
@@ -36,153 +37,204 @@ def __init__(
+        )
+        if self.feature_matrix_path:
+            feature_matrix.df.write.mode(self.session.write_mode).parquet(
+                self.feature_matrix_path
             )
-            if feature_matrix_path:
-                feature_matrix.df.write.mode(session.write_mode).parquet(
-                    feature_matrix_path
-                )
-            predictions.df.write.mode(session.write_mode).parquet(predictions_path)
-            session.logger.info(predictions_path)
-        elif (
-            run_mode == "train"
-            and gold_standard_curation_path
-            and gene_interactions_path
+        predictions.df.write.mode(self.session.write_mode).parquet(
+            self.predictions_path
+        )
+        self.session.logger.info(self.predictions_path)
+
+    def run_train(self) -> None:
+        """Run the training step.
+
+        Raises:
+            ValueError: If gold_standard_curation_path, gene_interactions_path, or wandb_run_name are not set.
+        """
+        if not (
+            self.gold_standard_curation_path
+            and self.gene_interactions_path
+            and self.wandb_run_name
+            and self.model_path
         ):
-            # Process gold standard and L2G features
-            gs_curation = session.spark.read.json(gold_standard_curation_path)
-            interactions = session.spark.read.parquet(gene_interactions_path)
-            study_locus_overlap = StudyLocus(
-                # We just extract overlaps of associations in the gold standard. This parsing is a duplication of the one in the gold standard curation,
-                # but we need to do it here to be able to parse gold standards later
-                _df=credible_set.df.join(
-                    f.broadcast(
-                        gs_curation.select(
-                            StudyLocus.assign_study_locus_id(
-                                f.col("association_info.otg_id"),  # studyId
-                                f.concat_ws(  # variantId
-                                    "_",
-                                    f.col("sentinel_variant.locus_GRCh38.chromosome"),
-                                    f.col("sentinel_variant.locus_GRCh38.position"),
-                                    f.col("sentinel_variant.alleles.reference"),
-                                    f.col("sentinel_variant.alleles.alternative"),
-                                ),
-                            ).alias("studyLocusId"),
-                        )
-                    ),
-                    "studyLocusId",
-                    "inner",
-                ),
-                _schema=StudyLocus.get_schema(),
-            ).find_overlaps(studies)
-
-            gold_standards = L2GGoldStandard.from_otg_curation(
-                gold_standard_curation=gs_curation,
-                v2g=v2g,
-                study_locus_overlap=study_locus_overlap,
-                interactions=interactions,
+            raise ValueError(
```

Comment:
Not sure I understand your comment. This is already asserting mandatory variables at the start of each method. I do it inside each function because train/test have different requirements